Board.java

```diff
@@ -0,0 +1,222 @@
+import java.util.HashSet;
+import java.util.Scanner;  
+import java.util.Set;
+
+class Board{ 
+    
+    public class Point{
+        int x, y;
+        Point(int x, int y){
+            this.x = x;
+            this.y = y;
+        }
+        
+        @Override
+        public String toString(){
+            return "["+x+", "+y+"]";
+        }
+        
+        @Override
+        public boolean equals(Object o){
+            return o.hashCode()==this.hashCode();
+        }
+
+        @Override
+        public int hashCode() {
+            return Integer.parseInt(x+""+y);
+        }
+    }
+    
+    private final char[][] board;
+    int WScore, BScore, remaining;
+    private final char boardX[] = new char[]{'A','B','C','D','E','F','G','H'};
+    
+    public Board(){ 
+        board = new char[][]{
+            {'_','_','_','_','_','_','_','_',},
+            {'_','_','_','_','_','_','_','_',},
+            {'_','_','_','_','_','_','_','_',},
+            {'_','_','_','W','B','_','_','_',},
+            {'_','_','_','B','W','_','_','_',},
+            {'_','_','_','_','_','_','_','_',},
+            {'_','_','_','_','_','_','_','_',},
+            {'_','_','_','_','_','_','_','_',},
+        };  
+    }
+    
+    private void findPlaceableLocations(char player, char opponent, HashSet<Point> placeablePositions){ 
+        for(int i=0;i<8;++i){
+            for(int j=0;j<8;++j){
+                if(board[i][j] == opponent){
+                    int I = i, J = j;  
+                    if(i-1>=0 && j-1>=0 && board[i-1][j-1] == '_'){ 
```

Comment:
instead of writing `i-1 >= 0` or `j <= n`, you should create a `isValid()` function and use that for all such purposes to keep the code cleaner

Reply:
similarly, an `isEmpty()`, `isOpponent()` and `isSelf()` function can be created, which in themselves use the `isValid()` function, should be used to replace the whole condition expression.